components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/jwks/JwksEndpoint.java

```diff
@@ -117,16 +123,30 @@ private String buildResponse(Map<String, Certificate> certificates)
         // If we read different algorithms from identity.xml then put them in a list.
         List<JWSAlgorithm> diffAlgorithms = findDifferentAlgorithms(accessTokenSignAlgorithm, config);
         // Create JWKS for different algorithms using new KeyID creation method.
-        for (Map.Entry certificateWithAlias : certificates.entrySet()) {
+        for (Map.Entry certChainInfoWithAlias : certChainInfosWithAliases.entrySet()) {
             for (JWSAlgorithm algorithm : diffAlgorithms) {
-                Certificate cert = (Certificate) certificateWithAlias.getValue();
-                String alias = (String) certificateWithAlias.getKey();
-                RSAPublicKey publicKey = (RSAPublicKey) cert.getPublicKey();
-                RSAKey.Builder jwk = new RSAKey.Builder(publicKey);
+                List<Base64> certList = new ArrayList<>();
+                String alias = (String) certChainInfoWithAlias.getKey();
+                X509Certificate cert =
+                        (X509Certificate) ((CertificateChainInfo) certChainInfoWithAlias.getValue()).getCertificate();
+                Certificate[] certChain =
+                        ((CertificateChainInfo) certChainInfoWithAlias.getValue()).getCertificateChain();
+                RSAKey.Builder jwk = new RSAKey.Builder((RSAPublicKey) cert.getPublicKey());
+                for (Certificate certificate : certChain) {
+                    try {
+                        certList.add(Base64.encode(certificate.getEncoded()));
+                    } catch (CertificateEncodingException exception) {
+                        String errorMessage = "Unable to encode the public certificate with alias: " + alias +
+                                " in the tenant domain: " + getTenantDomain();
+                        throw new CertificateEncodingException(errorMessage, exception);
```

Comment:
Would it be cleaner if we move cert chain building logic to a separate private method?